server/src/main/java/io/crate/execution/engine/aggregation/impl/CountAggregation.java

```diff
@@ -237,43 +246,78 @@ public MutableLong removeFromAggregatedState(RamAccounting ramAccounting,
+                    (ramAccounting, memoryManager, minNodeVersion) -> {
+                        ramAccounting.addBytes(LongStateType.INSTANCE.fixedSize());
+                        return new MutableLong(0L);
+                    },
+                    (values, state) -> state.add(1L)
+                );
+            default:
+                return null;
+        }
+    }
+
+    @Nullable
     @Override
-    public DocValueAggregator<?> getDocValueAggregator(List<DataType<?>> argumentTypes,
-                                                       List<MappedFieldType> fieldTypes,
+    public DocValueAggregator<?> getDocValueAggregator(List<Symbol> aggregationReferences,
+                                                       java.util.function.Function<List<String>, List<MappedFieldType>> getMappedFieldTypes,
+                                                       DocTableInfo table,
                                                        List<Literal<?>> optionalParams) {
-        if (argumentTypes.size() == 1) {
-            switch (argumentTypes.get(0).id()) {
-                case ByteType.ID:
-                case ShortType.ID:
-                case IntegerType.ID:
-                case LongType.ID:
-                case TimestampType.ID_WITH_TZ:
-                case TimestampType.ID_WITHOUT_TZ:
-                case FloatType.ID:
-                case DoubleType.ID:
-                case GeoPointType.ID:
-                    return new SortedNumericDocValueAggregator<>(
-                        fieldTypes.get(0).name(),
-                        (ramAccounting, memoryManager, minNodeVersion) -> {
-                            ramAccounting.addBytes(LongStateType.INSTANCE.fixedSize());
-                            return new MutableLong(0L);
-                        },
-                        (values, state) -> state.add(1L)
-                    );
-                case IpType.ID:
-                case StringType.ID:
-                    return new BinaryDocValueAggregator<>(
-                        fieldTypes.get(0).name(),
-                        (ramAccounting, memoryManager, minNodeVersion) -> {
-                            ramAccounting.addBytes(LongStateType.INSTANCE.fixedSize());
-                            return new MutableLong(0L);
-                        },
-                        (values, state) -> state.add(1L)
-                    );
-                default:
-                    return null;
+        if (aggregationReferences.size() != 1) {
+            return null;
+        }
+        if (aggregationReferences.get(0).valueType().id() == ObjectType.ID) {
+            //if count on object Type, optimize to count on one of the not-null-subcolumns
+            var allNotNullSubColumns = new ArrayList<>(table.notNullColumns());
+            var aggregationRef = (Reference) aggregationReferences.get(0);
+            for (var notNullCol : allNotNullSubColumns) {
+                //in FCFS basis
```

Comment:
What does this mean?

Reply:
sorry, made it more clear
`// the first seen not-null sub-column will be used`